### DISCUSSION TITLE
Wrong colors in neovim running within tmux with TERM set to xterm-256color

### DISCUSSION THREAD

**Discussion by lowstar** [ghostty-org/ghostty#3897] [Q&A] [answered]
Hi,

Is there any available workaround to the issue when neovim uses the wrong colors when run under tmux with TERM overridden to xterm-256color? Sometimes I need to override TERM from default xterm-ghostty when I work over ssh. When neovim is run directly (not within tmux session), colors work as expected even with TERM overriden. 

### Running neovim without tmux
<img width="917" alt="Screenshot 2024-12-29 at 16 07 15" src="https://github.com/user-attachments/assets/d621f6fa-37db-48ff-95f2-ceb48c722412" />

### Neovim running under tmux
<img width="917" alt="Screenshot 2024-12-29 at 16 07 37" src="https://github.com/user-attachments/assets/f7724388-5c63-4e02-8016-44985f16c478" />

regards,
Michal

**Comment by juanibiapina** [comment, marked as answer]
I have no issues with a similar setup. I have this line in tmux that seems to toggle this behavior:
```
set -ga terminal-overrides ",xterm-256color:Tc"
```

**Reply by lowstar** [reply]
> I have no issues with a similar setup. I have this line in tmux that seems to toggle this behavior:
> 
> ```
> set -ga terminal-overrides ",xterm-256color:Tc"
> ```

Seems like this fixed this issue. Could you clarify what "Tc" does exactly?

**Reply by lowstar** [reply]
> > I have no issues with a similar setup. I have this line in tmux that seems to toggle this behavior:
> > ```
> > set -ga terminal-overrides ",xterm-256color:Tc"
> > ```
> 
> Seems like this fixed this issue. Could you clarify what "Tc" does exactly?

It is described directly in tmux man page:

```
Tc     Indicate that the terminal supports the ‘direct colour’ RGB escape sequence (for example, \e[38;2;255;255;255m).

       If supported, this is used for the initialize colour escape sequence (which may be enabled by adding the ‘initc’ and ‘ccc’ capabilities to the
       tmux terminfo(5) entry).

       This is equivalent to the RGB terminfo(5) capability.
```
